github_pr.py

```diff
@@ -179,19 +184,38 @@ def _check_owner_cannot_ship(owner, comment_users):
 
 
 def _merge_pr(pr, **args):
-    if args['mergecondition_approved_mergers'] or args['mergecondition_approved_mergers_file'] or args['mergecondition_owner_cannot_ship']:
+    if args['condition_approved_mergers'] or args['condition_approved_mergers_file'] or args['condition_owner_cannot_ship']:
         merge_comment_users = [comment.user.login for comment in _load_issue(**args).get_comments().reversed if re.search(".*%s.*" % args['mergecomment'], comment.body)]
-        if args.get('mergecondition_approved_mergers'):
-            merge_comment_users = _check_approved_mergers(args['mergecondition_approved_mergers'], merge_comment_users)
-        if args.get('mergecondition_approved_mergers_file'):
-            merge_comment_users = _check_approved_mergers_file(args['mergecondition_approved_mergers_file'], merge_comment_users)
-        if args.get('mergecondition_owner_cannot_ship'):
+        if args.get('condition_approved_mergers'):
+            merge_comment_users = _check_approved_mergers(args['condition_approved_mergers'], merge_comment_users)
+        if args.get('condition_approved_mergers_file'):
+            merge_comment_users = _check_approved_mergers_file(args['condition_approved_mergers_file'], merge_comment_users)
+        if args.get('condition_owner_cannot_ship'):
             merge_comment_users = _check_owner_cannot_ship(pr.user.login, merge_comment_users)
         if merge_comment_users:
             pr.merge()
     else:
         pr.merge()
 
+def github_check_condition(**args):
+    check_required_fields(['token', 'repo', 'number'], **args)
+    gh = Github(args['token'])
+    repo = gh.get_repo(args['repo'])
+    pr = repo.get_pull(args['number'])
+    issue = _load_issue(**args)
+    last_commit_time = pytz.utc.localize(datetime.strptime(pr.get_commits().reversed[0].commit.raw_data['committer']['date'], '%Y-%m-%dT%H:%M:%SZ')).astimezone(pytz.timezone('US/Eastern'))
+    merge_comment_users = [comment.user.login for comment in issue.get_comments(since=last_commit_time).reversed if re.search('.*%s.*' % args['mergecomment'], comment.body)]
+
+    if not merge_comment_users:
+        raise NoMergeCommentError("There are no merge comments associated with this PR")
+
+    if args.get('condition_owner_cannot_ship'):
+        merge_comment_users = _check_owner_cannot_ship(pr.user.login, merge_comment_users)
+    if args.get('condition_approved_mergers_file'):
+        merge_comment_users = _check_approved_mergers_file(args['condition_approved_mergers_file'], merge_comment_users)
+
+    return merge_comment_users
```

Comment:
how are you planning on using this? if this function fails to throw an exception, the script will pass and you know you're good to go. Otherwise, it'll error non-zero and you'll know that it is lacking something? This will eventually post statuses back to the PR and then the merge button will be available based on that?